src/Message/FindCustomerRequest.php

```diff
@@ -16,12 +19,18 @@ public function getData()
     /**
      * Send the request with specified data
      *
-     * @param  mixed $data The data to send
-     * @return CustomerResponse
+     * @param mixed $data
+     *
+     * @return \Omnipay\Braintree\Message\CustomerResponse|\Omnipay\Common\Message\ResponseInterface
+     * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
      */
     public function sendData($data)
     {
-        $response = $this->braintree->customer()->find($this->getCustomerId());
+        try {
+            $response = $this->braintree->customer()->find($this->getCustomerId());
+        } catch (NotFound $exception) {
+            throw new NotFoundHttpException($exception->getMessage());
```

Comment:
I wouldn't throw this error, but rather `Omnipay\Common\Exception\InvalidResponseException`. This error could cause the application to render a 404 page instead of an error page (for example Laravel converts the HttpExceptions to specific errors)

Reply:
Thanks for your comment. I agree with you, but the main reason I'm throwing not found exception is for those (like me), who wants to recreate payment profile if the existing one is invalid or not found, without showing the error. So I'm throwing the general not found here, catching it in my Billable trait and recreating the profile. And I used generic Not Found instead of Braintree's one to have the logic working with other gateways also.

Reply:
I think it's better then to create your own exception that extends the omnipay invalidresponse exception.

Reply:
I agree, the problem is that if I create may own in this package, I can't catch it generalized, because other payment gateway packages will not have that exception. That's why I used Symphony's one

Reply:
If you extend the InvalidResponse one, all gateway can catch that. You can just add specific logic for that exception.

Reply:
Yes, they can, but they can’t detect that it is Not Found exception, they will catch as a just Invalid Response

Reply:
Anyways if I will just remove that part, and leave as is, it will throw Braintree's exception, like in all other places, instead of Omnipay's Invalid Response. So I've only generalized this, to have Not Found behave like Symphony's Not Found

Reply:
Correct, but if they want to implement specific behavior for Braintree, they can also use the Exception.

Reply:
Agreed, but in that case they have to implement in their code something like `catch Braintree/NotFound $exception` and then `catch AnotherGateway\NotFound $exception` etc.. so I wrote it here to give ability for developers to catch just one `NotFoundHttpException` for all gateways they may have

Reply:
The best way would be if Omnipay would have it's own NotFoundException extended from InvalidResponseException, and all gateways could through it in such cases, to differentiate all Invalid exceptions from NotFound

Reply:
We could do that perhaps. You can create a PR on omnipay/common if you want.

Reply:
OK, thank you, let me do it just now please. And update this one also

Reply:
Created, here is the PR for Omnipay Common https://github.com/thephpleague/omnipay-common/pull/187 and I've updated this one as well

Reply:
Can you remove the try/catch for now and leave it as before, so we can merge this? We can update it once we merge the PR and a new tag is released.

Reply:
OK, just 2 questions before I remove:
1. When you think the tag will be released?
2. May I leave the previous version with Symphony's NotFound until the release of Common package? To be able to handle it without hardcoding "Braintree" namespace in my application

Reply:
1. probably with a week or so, but not sure.
2. No but you could use a gateway specific exception.

Reply:
OK, done